components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -506,6 +507,22 @@ public static String getApplicationPermissionPath() {
 
     }
 
+    /**
+     * Validate application name according to the regex
+     *
+     * @return validated or not
+     */
+    public static boolean isRegexValidated(String applicationName) {
```

Comment:
You can reuse util methods from https://github.com/wso2/carbon-identity-framework/blob/5d4a622d69937b2032dc975fbe87eec4e543c7a8/components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityValidationUtil.java#L27-L27

instead of writing a new one.